### DISCUSSION TITLE
Cannot use "useHttpAttributeNameAsOperationId" after migration to NSwag 14

### DISCUSSION THREAD

**Discussion by alex1ozr** [RicoSuter/NSwag#4695] [Q&A] [answered]
Hello!
I'm currently updating my project to .NET 8 with the latest version of NSwag (14.0.0). I am encountering an issue with configuring documentGenerator/**aspNetCoreToOpenApi** section in nswag.json.
I cannot set "useHttpAttributeNameAsOperationId": true.

I have ASP.NET API (.NET 8) with controller methods like this:
```cs
[ProducesResponseType(typeof(PersonResponse), StatusCodes.Status201Created)]
[ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
[ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status409Conflict)]
[HttpPost("{id:guid}", Name = "CreatePerson")]
public async Task<IActionResult> Create([FromRoute] Guid id, [FromBody] PersonRequest request)
{ 
//...
 }
```
I also have  [ClientGenerator](https://github.com/alex1ozr/DockerTests/pull/3/files#diff-dc658f928431582580a6dafdc641884468d4d3cf3d8e3d33cfb68bb69318f0db) project with the following .csproj:
```cs
<Project Sdk="Microsoft.NET.Sdk">
    
    <PropertyGroup>
        <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
    </PropertyGroup>
    
    <Target Name="NSwag" AfterTargets="AfterBuild" Condition=" '$(Configuration)' == 'Debug' ">
        <Exec Command="$(NSwagExe_Net80) run nswag.json /variables:Configuration=$(Configuration)" 
              EnvironmentVariables="SkipMigration=True"
              ConsoleToMSBuild="true" />
    </Target>
    
    <ItemGroup>
        <PackageReference Include="NSwag.MSBuild" >
            <PrivateAssets>all</PrivateAssets>
            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
        </PackageReference>
    </ItemGroup>
    
    <ItemGroup>
      <ProjectReference Include="..\Api\Api.csproj" />
    </ItemGroup>

    <ItemGroup>
        <None Include="nswag.json" />
    </ItemGroup>
    
</Project>
```

Here is an example of my [nswag.json](https://github.com/alex1ozr/DockerTests/pull/3/files#diff-540ea8f66fc1f389eafbbc8c5bc9b0e80cf21c6e31f042877459dc75cbdb04b0):
```cs
{
  "runtime": "Net80",
  "defaultVariables": null,
  "documentGenerator": {
    "aspNetCoreToOpenApi": {
      "project": "../Api/Api.csproj",
      "documentName": "v1",
      "msBuildProjectExtensionsPath": null,
      "configuration": null,
      "runtime": null,
      "targetFramework": null,
      "noBuild": true,
      "msBuildOutputPath": null,
      "useHttpAttributeNameAsOperationId": true,
      "verbose": true,
      "workingDirectory": null,
      "aspNetCoreEnvironment": null,
      "output": "../Client/swagger.json",
      "newLineBehavior": "Auto"
    }
  },
  "codeGenerators": {
    //...
    }
  }
}
```

My goal is to get a generated client with the method CreatePerson instead of NSwag's default Person_Create.
Everything was fine till .NET 8...

Now, every time I build a ClientGenerator project it automatically rewrites nswag.json and deletes many (written by me) settings from there.

Okay, I tried different nswag.json samples from this repo by adding useHttpAttributeNameAsOperationId:
- A [big one](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.Sample.NET80/nswag.json) wasn't overridden by NSWag
- A [small one](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.Sample.NET80Minimal/nswag.json) was overridden by NSwag

In both cases, there was no success and I still got method names like {controllerName}_{controllerMethodName} instead of the expected ones.

Is this behaviour expected or is it a bug?
What's happened with useHttpAttributeNameAsOperationId setting in 14.0.0?

Thank you in advance for your help.

**Comment by krzyhan** [comment, marked as answer]
This occurs as I guess there is no property in command anymore.
https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.Commands/Commands/Generation/AspNetCore/AspNetCoreToOpenApiCommand.cs

SImilar to eq. `ApiGroupNames` - you can define it in your code 

```c#
 services.AddOpenApiDocument(document =>
 {
     document.UseHttpAttributeNameAsOperationId = true;
 });

```

https://github.com/RicoSuter/NSwag/issues/4524
![image](https://github.com/RicoSuter/NSwag/assets/20922676/61203e4f-2827-41e0-b64c-e4c6d476da5e)

**Reply by alex1ozr** [reply]
Thank you!
That really works :)
